samples/collect_data/collect_data.cpp

```diff
@@ -0,0 +1,196 @@
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2021 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+// Application for testing non-CPU-intensive features of BOINC.
+// TODO: make it exit after a certain amount of time
+
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <chrono>
+#include <thread>
+#include <sockpp/socket.h>
+#include <sockpp/tcp_connector.h>
+#if !_WIN32
+    #include <sockpp/unix_connector.h>
+#endif
+
+void my_sleep(std::chrono::nanoseconds nanoseconds) {
+    std::this_thread::sleep_for(nanoseconds);
+}
+
+sockpp::connector* get_connected(sockpp::connector* array_connection[], int arrSize) {
+    if (NULL == array_connection || 0 == arrSize) {
```

Comment:
If you change `arrSize` to `unsigned int` or `size_t`, this check will be valid. Now it will lead to an exception if negative value is passed, and it will pass this check since negative value != 0

Reply:
done.